tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -245,6 +245,14 @@ extern template void FillHelper::Exec(unsigned int, const std::vector<int> &, co
 extern template void
 FillHelper::Exec(unsigned int, const std::vector<unsigned int> &, const std::vector<unsigned int> &);
 
+template<typename T>
+void SetDirZero(T &){}
+
+inline void SetDirZero(TH1 &h)
+{
+   h.SetDirectory(nullptr);
+}
+
```

Comment:
The template version `SetDirZero(T &)` will always have precedence over `SetDirZero(TH1 &h)` unless you really pass a reference to `TH1` (not a reference to `TH1D`, for example). I don't think this is the desired behavior..?

(also I'd suggest `UnsetDirectory` as an alternative function name)

Reply:
Thanks for spotting this.